pkg/types/baremetal/validation/platform.go

```diff
@@ -405,6 +405,11 @@ func ValidateProvisioning(p *baremetal.Platform, n *types.Networking, fldPath *f
 			allErrs = append(allErrs, field.Invalid(fldPath.Child("clusterProvisioningIP"), p.ClusterProvisioningIP, fmt.Sprintf("%q is not in the provisioning network", p.ClusterProvisioningIP)))
 		}
 
+		// Ensure provisioningNetworkCIDR does not have any host bits set
+		if(string(p.ProvisioningNetworkCIDR.IP) != string(p.ProvisioningNetworkCIDR.IP.Mask(p.ProvisioningNetworkCIDR.Mask))){
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("provisioningNetworkCIDR"), p.ProvisioningNetworkCIDR, fmt.Sprintf("%q has host bits set", p.ProvisioningNetworkCIDR)))
```

Comment:
Just to give users the best possible chance of understanding how to fix this, could we modify the output to say something like:

```
172.22.0.2/24 has host bits set, should be e.g 172.22.0.0/24
```

When described that way, it's clear exactly what host bits refers to, and how to fix it?